tests/migrations/test_operations.py

```diff
@@ -1433,6 +1433,37 @@ def test_alter_field_pk(self):
         with connection.schema_editor() as editor:
             operation.database_backwards("test_alflpk", editor, new_state, project_state)
 
+    def test_alter_field_normal_column_to_pk_with_no_redundant_uk(self):
+        """
+        Tests the AlterField operation (not pk column -> pk) (#33125)
+        """
+        app_label = 'test_alflnormalpk'
+        project_state = self.apply_operations(app_label, ProjectState(), operations=[
+            migrations.CreateModel('Pony', fields=[
+                ('name', models.CharField(max_length=32, primary_key=True)),
+                ('nickname', models.CharField(max_length=32)),
+            ]),
+        ])
+        # Test the state alteration
+        first_operation = migrations.AlterField("Pony", "name", models.CharField(max_length=32))
+        second_operation = migrations.AlterField("Pony", "nickname", models.CharField(max_length=32, primary_key=True))
+        new_state = project_state.clone()
+        first_operation.state_forwards("test_alflnormalpk", new_state)
+        second_operation.state_forwards("test_alflnormalpk", new_state)
+        self.assertFalse(
+            project_state.models['test_alflnormalpk', 'pony'].fields['nickname'].primary_key
+        )
+        self.assertTrue(
+            new_state.models['test_alflnormalpk', 'pony'].fields['nickname'].primary_key
+        )
+        with connection.schema_editor() as editor:
+            first_operation.database_forwards("test_alflnormalpk", editor, project_state, new_state)
+            second_operation.database_forwards("test_alflnormalpk", editor, project_state, new_state)
+
+        with connection.schema_editor() as editor:
+            second_operation.database_backwards("test_alflnormalpk", editor, new_state, project_state)
+            first_operation.database_backwards("test_alflnormalpk", editor, new_state, project_state)
```

Comment:
This is not a regression test because it works without the patch. It should be enough to the check the number of constraints in the `tests.schema.tests.SchemaTests.test_alter_not_unique_field_to_primary_key()` (added in 6dd4edb1b4f5441c5f543e29395039839c50d10b), e.g.
```diff
diff --git a/tests/schema/tests.py b/tests/schema/tests.py
index 9e23c4608e..e258fb5ef6 100644
--- a/tests/schema/tests.py
+++ b/tests/schema/tests.py
@@ -741,6 +741,8 @@ class SchemaTests(TransactionTestCase):
         with connection.schema_editor() as editor:
             editor.remove_field(Author, Author._meta.get_field('id'))
             editor.alter_field(Author, old_field, new_field, strict=True)
+        # Redundant unique constraint is not added.
+        self.assertEqual(len(self.get_constraints_for_column(Author, 'uuid')), 1)
 
     @isolate_apps('schema')
     def test_alter_primary_key_quoted_db_table(self):
```